pilot/cmd/pilot-agent/config/config.go

```diff
@@ -147,6 +148,14 @@ func getMeshConfig(fileOverride, annotationOverride, proxyConfigEnv string) (mes
 		mc = *annotationMesh
 	}
 
+	// If after resolving the Mesh config, tracing has been disabled , make sure there are no tracing bits set
+	// in the proxy configuration. Otherwise, the Envoy bootstrap config will contain the default tracing configuration
+	// and the Zipkin cluster, and Envoy will periodically do unnecessary DNS requests to resolve the Zipkin cluster addresses.
+	if !mc.EnableTracing || !mc.DefaultConfig.EnableTracing {
+		log.Info("Tracing is disabled")
+		mc.DefaultConfig.Tracing = nil
```

Comment:
could we just always set mc.DefaultConfig.Tracing to nil to start? Install time configuration of tracing should override anyway, I think.